muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -688,6 +697,216 @@ private GuiderStudentWorkspaceActivityRestModel toRestModel(GuiderStudentWorkspa
+   * Output: 204 (no content)
+   */
+  @DELETE
+  @Path("/student/{ID}/contactLog/{CONTACTLOGENTRYID}")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response archiveStudentContactLog(@PathParam("ID") Long userEntityId, @PathParam("CONTACTLOGENTRYID") Long contactLogEntryId) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    
+    userSchoolDataController.removeStudentContactLogEntry(dataSource, userEntity.defaultSchoolDataIdentifier(), contactLogEntryId);
+    
+    return Response.noContent().build();
+
+  }
+  
+  /**
+   * 
+   * POST mApi().guider.student.contactLog.comments.create(userEntityId, contactLogEntryId, payload)
+   * 
+   * payload: {
+   * commentDate: Date
+   * text: "plaaplaa";
+   * }
+   * 
+   * @param userEntityId
+   * @param entryId
+   * @param payload
+   * @return
+   */
+  @POST
+  @Path("/student/{ID}/contactLog/{ENTRYID}/comments")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response createContactLogEntryComment(@PathParam("ID") Long userEntityId, @PathParam("ENTRYID") Long entryId, StudentContactLogEntryCommentRestModel payload) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    BridgeResponse<StudentContactLogEntryCommentRestModel> response = userSchoolDataController.createStudentContactLogEntryComment(dataSource, userEntity.defaultSchoolDataIdentifier(), entryId, payload);
+    if (response.ok()) {
+      return Response.status(response.getStatusCode()).entity(response.getEntity()).build();
+    }
+    else {
+      return Response.status(response.getStatusCode()).entity(response.getMessage()).build();
+    }    
+  }
+  
+  /**
+   * PUT mApi().guider.student.contactLog.comments.update(userEntityId, entryId, commentId, payload)
+   * 
+   * payload: {
+   * text: "plaa", 
+   * commentDate: Date
+   * }
+   * 
+   * @param userEntityId
+   * @param entryId
+   * @param commentId
+   * @param payload
+   * @return
+   * 
+   * Output: Updated StudentContactLogEntryCommentRestModel
+   */
+  @PUT
+  @Path("/student/{ID}/contactLog/{ENTRYID}/comments/{COMMENTID}")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateStudentContactLogComment(@PathParam("ID") Long userEntityId, @PathParam("ENTRYID") Long entryId, @PathParam("COMMENTID") Long commentId, StudentContactLogEntryCommentRestModel payload) {
+    String dataSource = sessionController.getLoggedUserSchoolDataSource();
+    UserEntity userEntity = userEntityController.findUserEntityById(userEntityId);
+    
+    BridgeResponse<StudentContactLogEntryCommentRestModel> response = userSchoolDataController.updateStudentContactLogEntryComment(dataSource, userEntity.defaultSchoolDataIdentifier(), entryId, commentId, payload);
```

Comment:
Same as when updating a contact log entry; check that the creatorId in payload is the same as the id of the currently logged in user.